src/mewbot/io/discord.py

```diff
@@ -91,6 +160,54 @@ async def on_ready(self) -> None:
         """
         self._logger.info("%s has connected to Discord!", self.user)
 
+        await self.retrieve_old_message()
+
+    async def retrieve_old_message(self) -> None:
+        """
+        If a startup_queue_depth is set, then
+        """
+        if not self._startup_queue_depth:
+            return
+
+        # Might want to, instead, wait for a queue
+        if not self.queue:
+            return
+
+        self._logger.info("Retrieving %s old messages", self._startup_queue_depth)
+
+        # The aim is to build a list of the last five messages the bot would have seen if it was up
+        # - iterate over all the guilds the bot can see
+        # - then iterate over all the text channels in that guild
+        # - grab a number of messages equal to the queue depth
+        # - append them to a master list
+        # - sort on time in the master list
+        # - return the queue depth number of items from the sorted list
+        past_messages: List[discord.Message] = []
+
+        # Short cut for itterating over all guilds, then all channels
+        for channel in self.get_all_channels():
+            # Ignoring everything which is not a text channel - nothing to do with past voice
+            if not isinstance(channel, discord.channel.TextChannel):
+                continue
+
+            messages = await channel.history(limit=5).flatten()
+            past_messages.extend(messages)
+
+        # Sort the messages and put the last five on the wire
+        past_messages = sorted(
+            past_messages, key=lambda x: float(x.created_at.timestamp()), reverse=True
+        )
+
+        for _ in range(self._startup_queue_depth):
+            message = past_messages[_]
```

Comment:
I do not like using `_` as an actual variable.
Would something like a slice make this clearer?
```suggestion
        for message in past_messages[0:self._startup_queue_depth]:
```

Reply:
Clearer and possibly faster!